SqlNoDeserializationTest failures

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/3636/

**server**
```
Error Message
Failed to resolve key metadata: java.io.IOException: KEY FAILURE
Stacktrace
com.hazelcast.sql.HazelcastSqlException: Failed to resolve key metadata: java.io.IOException: KEY FAILURE
	at com.hazelcast.sql.impl.QueryUtils.toPublicException(QueryUtils.java:76)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:241)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:207)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:203)
	at com.hazelcast.sql.SqlService.execute(SqlService.java:144)
	at com.hazelcast.sql.misc.SqlNoDeserializationTest.testMember(SqlNoDeserializationTest.java:109)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.sql.impl.QueryException: Failed to resolve key metadata: java.io.IOException: KEY FAILURE
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:59)
	at com.hazelcast.sql.impl.calcite.parse.QueryParser.parse(QueryParser.java:78)
	at com.hazelcast.sql.impl.calcite.OptimizerContext.parse(OptimizerContext.java:135)
	at com.hazelcast.sql.impl.calcite.CalciteSqlOptimizer.prepare(CalciteSqlOptimizer.java:126)
	at com.hazelcast.sql.impl.SqlServiceImpl.prepare(SqlServiceImpl.java:295)
	at com.hazelcast.sql.impl.SqlServiceImpl.query0(SqlServiceImpl.java:276)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:228)
	... 16 more
Caused by: com.hazelcast.sql.impl.QueryException: Failed to resolve key metadata: java.io.IOException: KEY FAILURE
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:67)
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:51)
	at com.hazelcast.sql.impl.schema.map.sample.MapSampleMetadataResolver.resolve(MapSampleMetadataResolver.java:81)
	at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.getFieldMetadata(PartitionedMapTableResolver.java:235)
	at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.getHeapMapFields(PartitionedMapTableResolver.java:180)
	at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.createTable(PartitionedMapTableResolver.java:120)
	at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.getTables(PartitionedMapTableResolver.java:73)
	at com.hazelcast.sql.impl.schema.SqlCatalog.<init>(SqlCatalog.java:42)
	at com.hazelcast.sql.impl.SqlServiceImpl.prepare(SqlServiceImpl.java:293)
	... 18 more
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: java.io.IOException: KEY FAILURE
	at com.hazelcast.internal.serialization.impl.SerializationUtil.handleException(SerializationUtil.java:103)
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:214)
	at com.hazelcast.sql.impl.schema.map.sample.MapSampleMetadataResolver.resolve(MapSampleMetadataResolver.java:75)
	... 24 more
Caused by: java.io.IOException: KEY FAILURE
	at com.hazelcast.sql.misc.SqlNoDeserializationTest$PersonKey.readData(SqlNoDeserializationTest.java:240)
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:160)
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:106)
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:51)
	at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:44)
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:208)
	... 25 more
```
**client**

```
Error Message
Failed to resolve key metadata: java.io.IOException: KEY FAILURE
Stacktrace
java.lang.AssertionError: Failed to resolve key metadata: java.io.IOException: KEY FAILURE
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.sql.misc.SqlNoDeserializationTest.testClient(SqlNoDeserializationTest.java:149)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast-sql-core/src/test/java/com/hazelcast/sql/misc/SqlNoDeserializationTest.java
+++ b/hazelcast-sql-core/src/test/java/com/hazelcast/sql/misc/SqlNoDeserializationTest.java
@@ -17,6 +17,7 @@
 package com.hazelcast.sql.misc;
 
 import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.connection.ClientConnection;
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.SqlExecuteCodec;
 import com.hazelcast.client.impl.protocol.codec.SqlFetchCodec;
@@ -25,7 +26,6 @@ import com.hazelcast.config.Config;
 import com.hazelcast.config.InMemoryFormat;
 import com.hazelcast.config.MapConfig;
 import com.hazelcast.core.HazelcastInstance;
-import com.hazelcast.internal.nio.Connection;
 import com.hazelcast.nio.serialization.Portable;
 import com.hazelcast.nio.serialization.PortableFactory;
 import com.hazelcast.nio.serialization.PortableReader;
@@ -53,7 +53,6 @@ import java.io.IOException;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.UUID;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
@@ -83,9 +82,7 @@ public class SqlNoDeserializationTest extends SqlTestSupport {
 
     @Before
     public void before() {
-        member = factory.newHazelcastInstance(config());
-        factory.newHazelcastInstance(config());
-
+        member = factory.newInstances(config(), 2)[0];
         client = factory.newHazelcastClient(clientConfig());
 
         prepare();
@@ -149,10 +146,10 @@ public class SqlNoDeserializationTest extends SqlTestSupport {
 
         SqlClientService clientService = (SqlClientService) client.getSql();
 
-        Connection connection = clientService.getRandomConnection();
+        ClientConnection connection = clientService.getRandomConnection();
 
         // Get the first page through the "execute" request
-        QueryId queryId = QueryId.create(UUID.randomUUID());
+        QueryId queryId = QueryId.create(connection.getRemoteUuid());
 
         ClientMessage executeRequest = SqlExecuteCodec.encodeRequest(
                 SQL,

--- a/hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java
+++ b/hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java
@@ -195,8 +195,8 @@ public class SqlClientService implements SqlService {
     /**
      * For testing only.
      */
-    public Connection getRandomConnection() {
-        Connection connection = client.getConnectionManager().getRandomConnection(false);
+    public ClientConnection getRandomConnection() {
+        ClientConnection connection = client.getConnectionManager().getRandomConnection(true);
 
         if (connection == null) {
             throw rethrow(QueryException.error(

